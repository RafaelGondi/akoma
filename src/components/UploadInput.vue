<template>
	<div>
		<div
			class="image-upload-wrap text-center py-3 px-5"
			:class="{
				'drag-enter': isOnDragEnterState,
			}"
			@dragenter="dragEnterHandler"
			@drop="dropHandler"
			@dragover="dragOverHandler"
		>
			<b-form-file
				v-show="false"
				v-model="file"
				:accept="accept"
				ref="fileInput"
			/>
			<div class="drag-text px-3 py-5">
				<span v-if="!file">
					<img src="../assets/images/upload-icon.svg"/>
					<div class="mt-3 title">
						<span v-if="!isOnDragEnterState">
							Arraste o arquivo aqui ou
							<a
								href="javascript:void(0)"
								class="search-link font-weight-bold"
								@click="linkClick"
							>
								pesquise no seu computador
							</a>
						</span>
						<span v-else>
							Solte aqui o seu arquivo
						</span>
					</div>
				</span>
				<span v-else>
					<img src="../assets/images/file-icon.svg"/>
					<div class="mt-3 title">
						<span v-if="!isOnDragEnterState">
							<span>{{ file.name }}</span>
							<div>
								<small>
									<a
										href="javascript:void(0)"
										class="search-link font-weight-bold"
										@click="file = null"
									>
										Remover
									</a>
								</small>
							</div>
						</span>
						<span v-else>
							Solte aqui o seu arquivo
						</span>
					</div>
				</span>
			</div>
		</div>
	</div>
</template>

<script>
export default {
	props: {
		/**
		 * O conteÃºdo do arquivo upado.
		 */
		value: {
			type: Object,
			default: () => ({}),
			description: 'The value that represents the uploaded file.',
			required: true,
		},
		/**
		 * Tipo de arquivos aceitos (Ref.: https://cutt.ly/fgkC0nU)
		 */
		accept: {
			type: String,
			default: null,
			description: `The accept prop defines what kind of file is accepted by
				the current instance of UploadInput.
				Reference to valid options: https://www.iana.org/assignments/media-types/
			`,
		},
	},

	data() {
		return {
			file: _.isEmpty(this.value) ? null :  this.value,
			isOnDragEnterState: false,
		};
	},

	watch: {
		file(newValue) {
			/**
				* Evento utilizado para implementar o v-model.
				* @event input
				* @type {Event}
			*/
			this.$emit('input', newValue);
		},
	},

	methods: {
		dropHandler(ev) {
			this.isOnDragEnterState = false;

			// Prevent the browser default behavior (open the file)
			ev.preventDefault();

			if (ev.dataTransfer.items) {
				if (ev.dataTransfer.items[0].kind === 'file') {
					this.file = ev.dataTransfer.items[0].getAsFile();
				}
			}
		},
		dragOverHandler(ev) {
			ev.preventDefault();
		},
		dragEnterHandler() {
			this.isOnDragEnterState = true;
		},
		linkClick() {
			// A lot of nested operations are needed because b-form-file wraps the input tag with a div
			this.$refs['fileInput'].$el.childNodes[0].click();
		}
	},
};
</script>
<style lang="scss" scoped>
@import '../assets/sass/app.scss';

.title {
	@include titulo-3;
}

.search-link {
	color: $azul-sonic-base;
}

.search-link:hover {
	text-decoration: underline;
}

.file-upload {
  background-color: white;
  width: 600px;
  margin: 0 auto;
  padding: 20px;
}

.image-upload-wrap {
  margin-top: 20px;
  border: 2px dashed $cinza-4;
  position: relative;
	border-radius: 16px;
}

.drag-text {
  text-align: center;
}

.drag-text h3 {
  color: $cinza-6;
}

.drag-enter {
	background-color: $azul-sonic-light-2;
  border: 3px dashed $cinza-4;
}
</style>
