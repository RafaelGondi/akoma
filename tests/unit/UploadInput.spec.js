import { mount, createLocalVue } from '@vue/test-utils';
import lodash from 'lodash';
import BootstrapVue from 'bootstrap-vue';
import UploadInput from '../../src/components/UploadInput.vue';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

test('Component is mounted properly', () => {
	window._ = lodash;
	const wrapper = mount(UploadInput, {
		localVue,
		propsData: {
			value: {},
		}
	});

	expect(wrapper).toMatchSnapshot();
});
