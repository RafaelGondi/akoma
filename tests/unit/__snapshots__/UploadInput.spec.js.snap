// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is mounted properly 1`] = `
<div class="upload-input text-center px-3">
  <div class="text-center px-2 py-4">
    <div class="custom-file b-form-file" style="display: none;"><input type="file" class="custom-file-input" style="z-index: -5;"><label data-browse="Browse" class="custom-file-label"><span class="d-block form-file-text" style="pointer-events: none;">No file chosen</span></label></div>
    <div><svg width="103" height="96" viewBox="0 0 103 96" fill="none" xmlns="http://www.w3.org/2000/svg" style="transform: scale(1);">
        <path d="M43.6196 87.2393C67.7101 87.2393 87.2393 67.7101 87.2393 43.6196C87.2393 19.5292 67.7101 0 43.6196 0C19.5292 0 0 19.5292 0 43.6196C0 67.7101 19.5292 87.2393 43.6196 87.2393Z" fill="#F0F5FF"></path>
        <path d="M56.6196 93.2393C80.7101 93.2393 100.239 73.7101 100.239 49.6196C100.239 25.5292 80.7101 6 56.6196 6C32.5292 6 13 25.5292 13 49.6196C13 73.7101 32.5292 93.2393 56.6196 93.2393Z" stroke="#3A7EDF" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path>
        <path d="M74.0675 49.6196L56.6196 32.1718L39.1718 49.6196" stroke="#3A7EDF" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path>
        <path d="M56.6196 67.0675V32.1718" stroke="#3A7EDF" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path>
      </svg>
      <div class="upload-input__title-md mt-3">
        <div>
          Arraste o arquivo aqui ou
          <a href="javascript:void(0)" class="upload-input__search-link font-weight-bold">
            pesquise no seu computador
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
